test/github_push_retest_test.go

```diff
@@ -29,63 +71,49 @@ func TestGithubSecondPushRequestGitOpsComments(t *testing.T) {
+	numberOfStatus := 3
+	waitOpts := twait.Opts{
+		RepoName:        g.TargetNamespace,
+		Namespace:       g.TargetNamespace,
+		MinNumberStatus: numberOfStatus,
+		PollTimeout:     twait.DefaultTimeout,
+		TargetSHA:       g.SHA,
 	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			waitOpts := twait.Opts{
-				RepoName:        g.TargetNamespace,
-				Namespace:       g.TargetNamespace,
-				MinNumberStatus: tt.prNum,
-				PollTimeout:     twait.DefaultTimeout,
-				TargetSHA:       g.SHA,
-			}
-			if tt.comment == "/cancel pipelinerun-on-push branch:"+g.TargetNamespace {
-				g.Cnx.Clients.Log.Info("/test pipelinerun-on-push on Push Request before canceling")
-				_, _, err = g.Provider.Client.Repositories.CreateComment(ctx,
-					g.Options.Organization,
-					g.Options.Repo, g.SHA,
-					&github.RepositoryComment{Body: github.String("/test pipelinerun-on-push branch:" + g.TargetNamespace)})
-				assert.NilError(t, err)
-				err = twait.UntilPipelineRunCreated(ctx, g.Cnx.Clients, waitOpts)
-				assert.NilError(t, err)
-			}
-			g.Cnx.Clients.Log.Infof("%s on Push Request", tt.comment)
-			_, _, err = g.Provider.Client.Repositories.CreateComment(ctx,
-				g.Options.Organization,
-				g.Options.Repo, g.SHA,
-				&github.RepositoryComment{Body: github.String(tt.comment)})
-			assert.NilError(t, err)
-
-			g.Cnx.Clients.Log.Info("Waiting for Repository to be updated")
-			_, err = twait.UntilRepositoryUpdated(ctx, g.Cnx.Clients, waitOpts)
-			assert.NilError(t, err)
-
-			g.Cnx.Clients.Log.Infof("Check if we have the repository set as succeeded")
-			repo, err := g.Cnx.Clients.PipelineAsCode.PipelinesascodeV1alpha1().Repositories(g.TargetNamespace).Get(ctx, g.TargetNamespace, metav1.GetOptions{})
-			assert.NilError(t, err)
-			if tt.comment == "/cancel pipelinerun-on-push branch:"+g.TargetNamespace {
-				assert.Equal(t, repo.Status[len(repo.Status)-1].Conditions[0].Status, corev1.ConditionFalse)
-			} else {
-				assert.Equal(t, repo.Status[len(repo.Status)-1].Conditions[0].Status, corev1.ConditionTrue)
-			}
-
-			pruns, err = g.Cnx.Clients.Tekton.TektonV1().PipelineRuns(g.TargetNamespace).List(ctx, metav1.ListOptions{})
-			assert.NilError(t, err)
-			assert.Equal(t, len(pruns.Items), tt.prNum)
-		})
+	err = twait.UntilPipelineRunCreated(ctx, g.Cnx.Clients, waitOpts)
+	assert.NilError(t, err)
+
+	comment := "/cancel pipelinerun-on-push branch:" + g.TargetNamespace
+	g.Cnx.Clients.Log.Infof("%s on Push Request", comment)
+	_, _, err = g.Provider.Client.Repositories.CreateComment(ctx,
+		g.Options.Organization,
+		g.Options.Repo, g.SHA,
+		&github.RepositoryComment{Body: github.String(comment)})
+	assert.NilError(t, err)
+
+	g.Cnx.Clients.Log.Infof("Waiting for Repository to be updated still to %d since it has been canceled", numberOfStatus)
+	_, _ = twait.UntilRepositoryUpdated(ctx, g.Cnx.Clients, waitOpts) // don't check for error, because canceled is not success and this will fail
+	assert.NilError(t, err)
```

Comment:
As we are not checking error here can we remove this assert ?

Reply:
yes and I made the test better ...